db/db.js

```diff
@@ -379,6 +379,44 @@ async function getLocationData(locationid) {
 
 }
 
+
+//Retrieves the partcle door coreID corresponding to a locationid
+//This is required for ServerTest.js
+
+async function getDoorParticleCoreIDFromLocationID(locationid){
+    try{
+        const results = await pool.query("SELECT (door_particlecoreid) FROM locations WHERE locationid = $1", [locationid])
+        if(results == undefined){
+            console.log('Error: No row with the locationid exists')
+            return null
```

Comment:
The reason that I was wondering where it was used is because I was planning to check whether the caller was reasonably handling a `null` return value. So if you end up adding it to be used somewhere, please think about this too ;-)